mypy/types.py

```diff
@@ -1062,7 +1063,16 @@ def has_readable_member(self, name: str) -> bool:
         """
         return all((isinstance(x, UnionType) and x.has_readable_member(name)) or
                    (isinstance(x, Instance) and x.type.has_readable_member(name))
-                   for x in self.items)
+                   for x in self.relevant_items())
+
+    def relevant_items(self) -> List[Type]:
+        """Returns all Union items relevant to current context.  In particular,
```

Comment:
I think it would be better if you just wrote that this function removes None from unions when strict None checking is off. (And "context" is a technical term in most of mypy so that word threw me off for the first few milliseconds.)